### ISSUE TITLE
Speed up tests by using a shared eval cache

### ISSUE THREAD

**Issue by zmitchell** [flox/flox#1139]
#1094 added the ability to use the Nix eval cache during builds, offering a substantial speedup of all operations that involve builds. Many integration tests run in isolated `$HOME` environments, filling out their own copy of an eval cache while they run and discarding it at the end of the test.

We can substantially speed up any tests that do builds by reusing the eval cache across tests that aren't specifically testing the eval cache.

**Acceptance criteria:**
- Bring in a shared eval cache during test suite setup
- Provide a function that disables the shared eval cache so that specific tests can opt-out of sharing.

**Comment by tomberek** [comment]
https://github.com/flox/flox-installers/pull/210 does some of this for different reasons; it should show what kind up speed-up there is when re-using HOME.

https://github.com/flox/flox-installers/actions/runs/8195729693/job/22414572378#step:5:2131 shows a subset of the tests running. It's easier to see what tests are expensive with timestamps on. Also note that only a subset of the tests are run due to failures and oddities discovered when turning them on sharing: global manifest re-use, db locking errors, etc. The worst offender in terms of speed is when needing to work with multiple systems.
```
Thu, 07 Mar 2024 22:41:05 GMT
ok 64 'flox install' provides suggestions when package not found
Thu, 07 Mar 2024 22:43:11 GMT
ok 65 'flox install' doesn't provide duplicate suggestions for a multi-system environment
```

Also note that there are other relevant Nix caches that help, especially to avoid superfluous Nixpkgs downloads; the binary-cache-v6.sqlite, fetcher-cache-v1.sqlite, and tarball-cache.

**Comment by tomberek** [comment]
Oh, and I believe we are doing this already? https://github.com/flox/flox/blob/main/cli/tests/setup_suite.bash#L294

**Comment by ysndr** [comment]
we are sharing the eval cache between tests _and_ have a method to opt out per test / per file.
However afaik wr still have to eval/build everything at least once becaue we dont (gh) cache the eval cache between runs of the test suite.
therefore, the more _different_ packages we use, the longer the tet suite will take (possibly mitigated somewhat by running tests in parallel but thats a tangent).

**Comment by mkenigs** [comment]
Closing since we share the eval cache and have a way to opt out. If we want to use GH cache to cache the eval cache, we can open a separate ticket
